### PR TITLE
Basic support for oauth2

### LINKED ISSUE #45: Fitbit API has changed
I received this:

> Why you are receiving this notice: You have an application registered to use the Fitbit API at https://dev.fitbit.com and the following changes may affect your application.
> 1. Heart rate time series will now only include data points when Fitbit believes the device was being worn. This change affects all heart rate data ever captured by a Fitbit device. You might consider re-requesting heart rate data retrieved prior to July 25, 2015.
> 2. Your application should consider all numerical ids in the Fitbit API to be unsigned 64 bit integers. Many ids will cross the maximum 32 bit integer in the next couple of weeks.
> 3. OAuth 2.0 requests will be restricted to JSON and TCX formats. XML will no longer be supported with OAuth 2.0 requests starting August 17, 2015.
> 4. The custom logging endpoints (deprecated in October 2014) will be removed on August 27, 2015.
> API-Get-Heart-Rate (e.g. GET /1/user/-/heart/date/2010-02-21.json, replaced with heart rate time series)
> API-Log-Heart-Rate (e.g. POST /1/user/-/heart.json)
> API-Delete-Heart-Rate-Log (e.g. DELETE /1/user/-/heart/123.json)
> API-Get-Blood-Pressure (e.g. GET /1/user/-/bp/date/2010-02-21.json)
> API-Log-Blood-Pressure (e.g. POST /1/user/-/bp.json)
> API-Delete-Blood-Pressure-Log (e.g. DELETE /1/user/-/bp/123.json)
> API-Get-Glucose (e.g. GET /1/user/-/glucose/date/2010-02-21.json)
> API-Log-Glucose (e.g. POST /1/user/-/glucose.json)
> API-Get-Body-Measurements (e.g. GET /1/user/-/body/date/2010-02-21.json)
> API-Log-Body-Measurements (e.g. POST /1/user/-/body.json)
> All measurements except fat and weight will be removed from API-Get-Body-Measurements (e.g. GET /1/user/-/body/date/2010-02-21.json) and API-Log-Body-Measurements (e.g. POST /1/user/-/body.json)
> If you have questions, please post in the Web API support forum. If you have a confidential inquiry, please email us.

I think this gem is not working anymore, I cant dwonload any data from Fitbit API.

### LINKED ISSUE #48: Upgrade for OAuth 2.0
Just to bump the issue already raised with #44 and the switch isn't too far in the future. Is there any recommendation on which alternative Gem to use?

### PR DESCRIPTION
**Pull request by chrisdambrosio**
This removes oauth code that enables retrieving an access token with a
fitbit generated secret, which is not supported with Fitbit's oauth2
authentication. This change requires that the access token is generated
and passed to the Fitgem client.

This closes #45, and closes #48

### CODE REVIEW

**Comment by whazzmaster** [comment]
I just realized the test matrix includes unsupported ruby versions- could you update `.travis.yml` to 

```yml
rvm:
  - 2.1
  - 2.2
  - 2.3
```

And repush to trigger CI?

**Comment by chrisdambrosio** [comment]
I updated the Travis configuration to get CI going again; should be good to go now.

File: lib/fitgem/client.rb [hunk 1]
```diff
@@ -229,8 +148,7 @@ def post(path, body='', headers={})
       end
 
       def raw_post(path, body='', headers={})
-        headers.merge!('User-Agent' => "fitgem gem v#{Fitgem::VERSION}", 'Accept-Language' => @api_unit_system,
-                       'Accept-Locale' => @api_locale)
+        headers.merge! default_headers
         uri = "/#{@api_version}#{path}"
         access_token.post(uri, body, headers)
```
Comment (proby) [inline lib/fitgem/client.rb]: The signature of `#post` changed in the OAuth2 gem. It's now only 2 parameters: `def post(path, opts = {}, &block)` (https://github.com/intridea/oauth2/blob/master/lib/oauth2/access_token.rb#L120). The other methods likely have the same issue. All the request method parameters are the same in the OAuth2 gem (`(path, opts = {}, &block)`).

File: lib/fitgem/client.rb [hunk 2]
```diff
@@ -249,9 +166,17 @@ def raw_delete(path, headers={})
       def extract_response_body(response)
         return {} if response.nil?
 
-        raise ServiceUnavailableError if response.code == '503'
+        raise ServiceUnavailableError if response.status == 503
 
         response.body.nil? ? {} : JSON.parse(response.body)
```
Comment (proby) [inline lib/fitgem/client.rb]: In some live testing I did a couple days ago I was seeing the Fitbit API returning an empty string in the body, not `nil`. This was causing this line to attempt to `JSON.parse('')` which raises `JSON::ParserError: A JSON text must at least contain two octets!`

**Comment by albertpak** [comment]
is this going to be merged in any time soon?

**Comment by jmhooper** [comment]
^ I'd also like to see this merged.

**Comment by whazzmaster** [comment]
I'm going to rev the major version on this change before publishing. Also, I'm not a fan of Fitbit anymore and will not be maintaining the gem going forward. Looking for folks interested in taking over; I'll be updating the README shortly.

**Comment by dsjoerg** [comment]
Hey @chrisdambrosio is your code working?  Does it handle the refresh-token dance or somehow not have to?   I've got your code working for me, however the access token expires after an hour and I'm trying to be lazy and use someone else's solution if it exists :)  thanks in advance!

**Comment by chrisdambrosio** [comment]
@dsjoerg It does not handle refreshing the token.

I handle expired tokens by rescuing `OAuth2::Error` on Fitgem requests. If the error is `expired_token` then I call the code in my app that refreshes the token.

**Pull request merged**